<?xml version="1.0"?>
<launch>

  <arg name="map_file" default="/home/ssapl/testbot_ws/src/hw_t/map/sbpl.yaml"/>
  <arg name="model" default="$(find hw_t)/urdf/wheelcheck.urdf.xacro"/>
  <arg name="base_frame" default="base_plate"/>
  <arg name="move_forward_only" default="false"/>
  <arg name="lidar_port" default="/dev/lidar"/>
  <arg name="amcl_scan_topic" default="/scan"/>
  <!-- <arg name="amcl_scan_topic2" default="/scan_2"/> -->
  <!-- <arg name="lidar_ip" default="192.168.7.92"/> -->

  <!-- <node name="odo" pkg="bot" type="scanner.py"/> -->

  <!-- <include file="$(find diff_drive)/launch/ddrive_moons.launch"/> -->

  <!-- <include file="$(find bot)/launch/laser_filter.launch"/>   -->
  
    <include file="$(find hw_t)/launch/run.launch"></include>

  <!-- <include file="$(find diff_drive)/launch/ddrive_moons.launch"/> -->

  <!-- <arg name="lidar_ip" value="$(arg lidar_ip)"/> -->

  <!-- <node  pkg="bot" type="initalpose.py" name="pose_saver" output="screen"/>  -->
 
  <!-- <param name="robot_description" command="$(find xacro)/xacro $(arg model)" /> -->

  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" >
  <!-- </node>
   <node pkg="map_server" name="map_server1" type="map_server" args="/home/taurus1/testbot_ws/src/hw_t/map/sbpl2.yaml">
  <remap from='map' to="map2"/> -->
 </node>
  
  <!-- <include file="$(find ira_laser_tools)/launch/laserscan_multi_merger.launch" /> -->
  <!-- <node pkg="tf" type="static_transform_publisher" name="laser" args="0 0 0 0 0 0 1 /map /odom  30"/> -->
 <include file="$(find hw_t)/launch/amcl2.launch" >
 <arg name="scan_topic" value="$(arg amcl_scan_topic)"/>
 <!-- <! <arg name="scan_topic2" value="$(arg amcl_scan_topic2)"/>  -->
 </include>
 <!-- <include file="$(find hw_t)/launch/localization.launch"/> -->
 <!-- <include file="$(find als_ros)/launch/mcl.launch">

 </include> -->

  <include file="$(find hw_t)/launch/move_base.launch"> 
	<arg name="move_forward_only" value="$(arg move_forward_only)" />
  </include>
    <include file="$(find xsens_driver)/launch/xsens_driver.launch"/>  
    <!-- <include file="$(find hw_t)/launch/camera2.launch"/> -->
   <node name="ekf_localization" pkg="robot_localization" type="ekf_localization_node"  >
    <rosparam command="load" file="$(find hw_t)/config/ekf.yaml" />
    
    

     <remap from="odometry/filtered" to="odometry/filtered" /> 
 
    <remap from="accel/filtered" to="/accel/filtered" />
  </node>
  <!-- <include file="$(find hw_t)/launch/new_camera.launch"/> -->
     
    <!-- <node pkg="tf" type="static_transform_publisher" name="base_to_odom" args="0 0 0 0 0 0 /odom /base_link 100"/> -->
  <!-- <include file="$(find hw_t)/launch/robo_ekf.launch"/>  -->
    <!-- <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
  <param name="output_frame" value="odom_combined"/>
  <param name="base_footprint_frame" value="base_link"/>
  <param name="freq" value="30.0"/>
  <param name="sensor_timeout" value="1.0"/>  
  <param name="odom_used" value="true"/>
  <param name="imu_used" value="true"/>
  <param name="vo_used" value="false"/>
  <param name="publish_tf" value="true"/>
  <remap from="odom" to="odom"/>
  <remap from="imu_data" to="imu_data" />
</node> -->
  <!-- <node pkg="robot_localization" type="ukf_localization_node" name="ukf_se" clear_params="true">
    <rosparam command="load" file="$(find hw_t)/config/ukf.yaml" />

      Placeholder for output topic remapping
    <remap from="odometry/filtered" to="odom"/>
    <remap from="accel/filtered" to="odom/accel"/>  

  </node> -->

</launch>

 